src/v/datalake/schema_registry.cc

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "model/record.h"
+
+std::optional<uint32_t> get_value_schema_id(const model::record& record) {
```

Comment:
where is the format of the record defined? is this format defined by the schema registry? if so, perhaps this should be a helper that lives in schema registry?